### PR TITLE
Feature/backend cleanup

### PR DESCRIPTION
**Pull request by Nicolai-Bak**
Added dto's to make it possible to avoid object cycles. 
Added mapping functions to responseDtos. 
Cleaned up nullable warnings

### CODE REVIEW

File: todo-frontend-angular/src/app/todo/SubtaskModel.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type SubtaskModel = {
```
Comment (PRP-ITM) [inline todo-frontend-angular/src/app/todo/SubtaskModel.tsx]: this is still a `.tsx` file :)

File: todo-frontend-angular/src/app/todo/TodoModel.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+//import { SubtaskModel } from './SubtaskModel'
```
Comment (PRP-ITM) [inline todo-frontend-angular/src/app/todo/TodoModel.tsx]: Still a `.tsx` file

File: todo-frontend-angular/src/app/todo/SubtaskModel.tsx (hunk 1 above)
Comment (Nicolai-Bak) [reply todo-frontend-angular/src/app/todo/SubtaskModel.tsx]: updated

File: todo-frontend-angular/src/app/todo/TodoModel.tsx (hunk 2 above)
Comment (Nicolai-Bak) [reply todo-frontend-angular/src/app/todo/TodoModel.tsx]: updated

**Review by AntonEsbjergIT-Minds — APPROVED** [review]
Looks good, seems ready to merge

**Pull request closed without merging**